streams/src/test/java/org/apache/kafka/streams/state/internals/CompositeReadOnlyWindowStoreTest.java

```diff
@@ -106,17 +113,37 @@ public void shouldNotGetValuesFromOtherStores() throws Exception {
         assertEquals(Collections.singletonList(new KeyValue<>(1L, "my-value")), results);
     }
 
-
     @Test(expected = InvalidStateStoreException.class)
     public void shouldThrowInvalidStateStoreExceptionOnRebalance() throws Exception {
         final CompositeReadOnlyWindowStore<Object, Object> store = new CompositeReadOnlyWindowStore<>(new StateStoreProviderStub(true), QueryableStoreTypes.windowStore(), "foo");
         store.fetch("key", 1, 10);
     }
 
-    @Test(expected = InvalidStateStoreException.class)
+    @Test
     public void shouldThrowInvalidStateStoreExceptionIfFetchThrows() throws Exception {
         underlyingWindowStore.setOpen(false);
-        underlyingWindowStore.fetch("key", 1, 10);
+        final CompositeReadOnlyWindowStore<Object, Object> store =
+                new CompositeReadOnlyWindowStore<>(stubProviderOne, QueryableStoreTypes.windowStore(), "window-store");
+        try {
+            store.fetch("key", 1, 10);
+            Assert.fail("InvalidStateStoreException was expected");
+        } catch (InvalidStateStoreException e) {
+            Assert.assertEquals("State store is not available anymore and may have been migrated to another instance; " +
+                    "please re-discover its location from the state metadata.", e.getMessage());
+        }
+    }
+
+    @Test
+    public void testWindowStoreIterator() throws Exception {
```

Comment:
Can you split this out into 2 tests please?. One for each method that is being tested. Thanks